net: mqtt_sn: Fix several minor issues

rlubos opened the pull request:
A set of minor fixes for the MQTT SN library and dependencies. For details, please check commit messages.

Fixes #89432

rlubos committed:
net: mqtt_sn: Make sure multicast functionalities are enabled

UDP transport for the MQTT SN libraries depends on IGMP/MLD APIs
unconditionally (via respective setsockopt calls) and without them being
enabled transport initialization would fail. Therefore, ensure
respective multicast libraries are always enabled if MQTT SN UDP
transport is used.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
net: mqtt_sn: Verify result of transport initialization

The result of the transport init() function should be propagated to the
application, otherwise the initialization could fail silently.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
net: mqtt_sn: Use zsock_inet_ntop instead of inet_ntop

Use zsock_inet_ntop() instead of inet_ntop() to avoid dependency to the
POSIX subsystem in the library.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
net: ip: igmp: Ensure IGMP APIs work with offloaded interfaces

IGMP APIs are commonly used across the codebase to configure IPv4
multicast addresses on network interfaces. Sending IGMP reports however
works only for native interfaces as it uses low-level APIs. Therefore,
in order to make the APIs at least semi-functional for offloaded
interfaces as well (i.e. allow to configure multicast address on
the interface), return early in case interface is offloaded.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos committed:
net: ip: mld: Ensure MLD APIs work with offloaded interfaces

MLD APIs are commonly used across the codebase to configure IPv6
multicast addresses on network interfaces. Sending MLD reports however
works only for native interfaces as it uses low-level APIs. Therefore,
in order to make the APIs at least semi-functional for offloaded
interfaces as well (i.e. allow to configure multicast address on
the interface), return early in case interface is offloaded.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

BeckmaR commented on subsys/net/lib/mqtt_sn/mqtt_sn_transport_udp.c:
```diff
@@ -25,11 +25,13 @@ static char *get_ip_str(const struct sockaddr *sa, char *s, size_t maxlen)
 	switch (sa->sa_family) {
 	case AF_INET:
-		inet_ntop(AF_INET, &(((struct sockaddr_in *)sa)->sin_addr), s, maxlen);
+		zsock_inet_ntop(AF_INET, &(((struct sockaddr_in *)sa)->sin_addr),
```
Maybe a stupid question and I'm on mobile, but if this removes a dependency, shouldn't a header include or a config be removed as well?

rlubos replied:
True, there was one redundant header include left in the file. We can remove POSIX requirement from the sample as well, will update.

rlubos committed:
samples: net: mqtt_sn_publisher: Drop POSIX requirement

The library no longer requires POSIX, therefore remove it from the
sample as well.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

mpecar reviewed (approved):
This fixes everything in #89432
